### ISSUE TITLE
Unsupported action type: '_ListAppend'

### ISSUE THREAD

**Issue by aczire** [pynamodb/PynamoDB#1036]
Hi,

I was trying to add an item to the list,

This is the created_on_list attribute.
`created_on_list= ListAttribute()`

I'm updating the list like this,
`                OrderMarker.created_on_list.append(
                    [marker_table.last_update_timestamp] ),`

When executed, I'm getting exception as shown below. Any idea?

`        "  File \"/opt/python/pynamodb/transactions.py\", line 118, in update\n    return_values_on_condition_failure=return_values\n",
        "  File \"/opt/python/pynamodb/models.py\", line 473, in get_update_kwargs_from_instance\n    return self._get_connection().get_operation_kwargs(hk_value, range_key=rk_value, key=KEY, actions=actions, condition=condition, return_values_on_condition_failure=return_values_on_condition_failure)\n",
        "  File \"/opt/python/pynamodb/connection/table.py\", line 83, in get_operation_kwargs\n    return_values_on_condition_failure=return_values_on_condition_failure\n",
        "  File \"/opt/python/pynamodb/connection/base.py\", line 924, in get_operation_kwargs\n    update_expression = Update(*actions)\n",
        "  File \"/opt/python/pynamodb/expressions/update.py\", line 85, in __init__\n    self.add_action(action)\n",
        "  File \"/opt/python/pynamodb/expressions/update.py\", line 97, in add_action\n    raise ValueError(\"unsupported action type: '{}'\".format(action.__class__.__name__))\n",
        "ValueError: unsupported action type: '_ListAppend'\n"`

Using Version: 5.1.0

**Comment by ikonst** [comment]
Can you show more complete code?

**Comment by aczire** [comment]
Its only the list append, if I remove that everything works fine. Seems I'm missing something somewhere!
`
class order_marker(Model):
    """
    Order Marker for lastupdated ts and seq.
    """
    class Meta:
        table_name = os.environ['ORDER_MARKER_TABLE']
    order_uniqueue_id = UnicodeAttribute(hash_key=True)
    order_id = UnicodeAttribute(null=True)
    last_update_timestamp = UnicodeAttribute(null=True)
    last_update_sequence = UnicodeAttribute(null=True)
    created_on_list = ListAttribute(default=list)

    def __eq__(self, other):
        return isinstance(other, order_marker) and \
            self.order_id == other.order_id

    def __repr__(self):
        return str(vars(self))
        
    @staticmethod
    def check_seen(marker_table):
        """
        Update the last update fields with the last modified timestamp.
        This will act as a checkpoint for de-duplication.
        """
        req_token = str(uuid.uuid4())

        try:
            with TransactWrite(connection=Connection(),
                               client_request_token=req_token) as transaction:
                epoch_time = str(int(time.time()))

                transaction.update(
                    marker_table,
                    actions=[
                        order_marker.order_id.set(marker_table.order_id),
                        order_marker.created_on_list.append(
                            [marker_table.last_update_timestamp]),
                        order_marker.last_update_timestamp.set(
                            marker_table.last_update_timestamp),
                        order_marker.last_update_sequence.set(
                            marker_table.last_update_sequence),
                    ],
                    condition=(
                        (order_marker.last_update_sequence !=
                         marker_table.last_update_sequence)
                    )
                )
        except TransactWriteError as ex:
            logger.error(
                "Failed to update last seen timestamps : " + str(ex.cause_response_code))
            raise ex from None
`

**Comment by ikonst** [comment]
Ack, so it's a bit contrived, but follows this:
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateItem.html#DDB-UpdateItem-request-UpdateExpression

tl;dr -
```diff
-order_marker.created_on_list.append([marker_table.last_update_timestamp]),
+order_marker.created_on_list.set(order_marker.created_on_list.append([marker_table.last_update_timestamp])),
```

:|

**Comment by aczire** [comment]
Thanks for the prompt response,
for some reason "order_marker.created_on_list.set(order_marker.created_on_list.append([marker_table.last_update_timestamp]))," didn't work.

But the below did!
```
order_marker.created_on_list.set(
    (order_marker.created_on_list | []).append(
        [marker_table.last_update_timestamp])),
order_marker.sequence_list.set(
    (order_marker.sequence_list | []).append(
        [marker_table.last_update_sequence])),
```

Thanks again for the prompt assistance!!! :)
